File: coin_flip.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define len(x)  (sizeof(x) / sizeof((x)[0]))

int main() {
	int n_simulation = 10000;
	int n_flips = 1000;
	int heads[n_simulation];
	int mean = 0;
	srand(time(NULL));

	for (int i=0; i<n_simulation; i++) {
		int h = 0;
		for (int j=0; j<n_simulation; j++) {
			int coin = rand() % 2;
			h += coin;
		}
		heads[i] = h;
	}
	for (int x=0; x<len(heads); x++) mean += heads[x];
	mean /= n_flips;
	printf("Probability of getting head: %f", (double)mean/n_flips);
}

File: coin_flip.py
import random
import numpy as np
import matplotlib.pyplot as plt

n_simulation = 1000
n_flips = 1000
av_vals = []

for i in range(n_simulation):
    val = 0
    values = []

    for j in range(n_flips):
        coin = random.randint(0,1)

        val += coin
        prob = val/(j+1)

        values.append(prob)

    x = np.linspace(0,n_flips,n_flips)
    plt.plot(x,values)
    av_vals.append(sum(values)/len(values))

print("Probability of getting head: {}%".format(sum(av_vals)/len(av_vals)))
plt.title("Flipping a coint - Montecarlo simulation")
plt.xlabel("Flips number")
plt.ylabel("Probabilty")
plt.axhline(0.5,0,n_flips,color="red")
plt.show()
